File: JavaScipt/EulerProblem0025.js
/*
 * The problem can't be solved using recursive
 *
 *function fibDigit(count, j, k) {
 *  var numberStr = k;
 *	numberStr = numberStr.toString();
 *	if (numberStr.length === 1000) {
 *		return count;
 *	} else {
 *		return fibDigit(count+1, k, j+k);
 *	}
 *}
 */

// Just ignore it, still can't be solve with brutefore looping too
function fibDigit(count, j, k) {
	var numberStr = k.toString();
	while (numberStr.length < 1000) {
		var temp = j;
		j = k;
		k += temp;
		numberStr = k.toString();
		count++;
	}
	return count;
}

console.log(fibDigit(2, 1, 1));

File: Python/EulerProblem0025.py
# The problem Can't be solved using recursive
#def fibDigit(count=2, j=1, k=1):
#	numberStr = str(k)
#	if len(numberStr) > 999:
#		return count, j, k
#	else:
#		return fibDigit(count+1, k, j+k)

def fibDigit(count=2, j=1, k=1):
	numberStr = str(k)
	while len(numberStr) < 1000:
		count += 1
		temp = j
		j = k
		k = temp+k
		numberStr = str(k)

	return count

print fibDigit();